holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
>``_augmentable_1d_index``

Don't really know what that means.

> ``_allows_implicit_1d_index``

Allows doesn't seem right, it's adding an index if you don't provide it.

> ``_auto_1d_indexable``

This comes closest out of your suggestions.